text/src/grep.rs

```diff
@@ -0,0 +1,448 @@
+//
+// Copyright (c) 2024 Hemi Labs, Inc.
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+extern crate clap;
+extern crate plib;
+
+use clap::Parser;
+use gettextrs::{bind_textdomain_codeset, textdomain};
+use plib::PROJECT_NAME;
+use regex::{Regex, RegexBuilder};
+use std::{
+    fs::File,
+    io::{self, BufRead, BufReader},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+/// grep - search a file for a pattern.
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about)]
+struct Args {
+    /// Match using extended regular expressions.
+    #[arg(short = 'E')]
```

Comment:
Our arg policy:    Support `long` options, for each arg.   Use the same --option as Linux grep where possible (because POSIX does not specify long options).